webapp/src/Service/ImportProblemService.php

```diff
@@ -524,11 +532,11 @@ public function importZippedProblem(
             $contestProblem->setProblem($problem);
             $contestProblem->setContest($contest);
             $this->em->persist($contestProblem);
+            $this->em->flush();
         }
 
-        $this->em->flush();
```

Comment:
Can you not move the flush from L530 here adn remove L535?

Reply:
No, in line 532 we rely that the newly persisted object has been flushed.